### PR TITLE
feat: [DHIS2-12296] support programRuleVariable type

### PR DESCRIPTION
**Pull request by Bekkalizer**

### CODE REVIEW

File: src/core_modules/capture-core-utils/rulesEngine/services/VariableService/VariableService.js [hunk 1]
```diff
@@ -325,12 +319,13 @@ export class VariableService {
             return null;
         }
 
-        const value = VariableService.getDataElementValueForVariable(dataElementValue, dataElementId, programVariable.useNameForOptionSet, sourceData.dataElements, sourceData.optionSets);
+        const value = VariableService.getDataElementValueForVariable(dataElementValue,
+            dataElementId, programVariable.useNameForOptionSet, sourceData.dataElements, sourceData.optionSets);
```
Comment (superskip) [inline src/core_modules/capture-core-utils/rulesEngine/services/VariableService/VariableService.js:323]: Split this line in a similar fashion as line 303?

File: src/core_modules/capture-core-utils/rulesEngine/services/VariableService/VariableService.js [hunk 2]
```diff
@@ -262,32 +262,20 @@ export class VariableService {
...
-                EMPTY_STRING,
-                typeKeys.TEXT, {
-                    variablePrefix: variablePrefixes.TRACKED_ENTITY_ATTRIBUTE,
-                    useNameForOptionSet: programVariable.useNameForOptionSet,
-                },
-            );
-        }
-
+    getVariableForSelectedEntityAttributes(programVariable: ProgramRuleVariable,
+        sourceData: SourceData): ?RuleVariable {
```
Comment (superskip) [inline src/core_modules/capture-core-utils/rulesEngine/services/VariableService/VariableService.js:273]: More consistency when you split long lines, please. I would prefer this format:
```
functionName(
   arg1: Type1,
   arg2: Type2,
): ReturnType {
   ...
}
```

File: src/core_modules/capture-core-utils/rulesEngine/services/VariableService/VariableService.js [hunk 3]
```diff
@@ -206,7 +206,7 @@ export class VariableService {
         return {
             variableValue: this.onProcessValue(value, type),
             useCodeForOptionSet: !useNameForOptionSet,
-            variableType: useNameForOptionSet ? typeKeys.TEXT : type,
+            variableType: type || typeKeys.TEXT,
```
Comment (superskip) [inline src/core_modules/capture-core-utils/rulesEngine/services/VariableService/VariableService.js:209]: Seems like a bold change to exclude `useNameForOptionSet` from the expression entirely, but maybe you see the reason for it being there in the first place?

**Review by superskip — APPROVED** [review]
Just a few stylistic nitpicks, seems fine otherwise :+1:

File: src/core_modules/capture-core-utils/rulesEngine/services/VariableService/VariableService.js (hunk 3 above)
Comment (Bekkalizer) [reply src/core_modules/capture-core-utils/rulesEngine/services/VariableService/VariableService.js:209]: This was an error actually, but its fixed by hard coding "TEXT" outside of the call to this function when there IS an option set AND nameForOptionSet == true. Fixed in this commit: https://github.com/dhis2/capture-app/pull/2560/commits/d0e00d779d5928b6b5528cee5fd903e98dd3396c

**Review by JoakimSM — APPROVED** [review]

**Pull request merged**